src/vs/workbench/contrib/preferences/browser/preferencesActions.ts

```diff
@@ -118,8 +120,9 @@ export class OpenGlobalKeybindingsAction extends Action {
 		super(id, label);
 	}
 
-	run(event?: any): Promise<any> {
-		return this.preferencesService.openGlobalKeybindingSettings(false);
+	run(args: any): Promise<any> {
+		const query = typeof args === 'string' ? args : undefined;
+		return this.preferencesService.openGlobalKeybindingSettings(false, query);
```

Comment:
I would suggest to remove this action and contribute the same command like below

https://github.com/microsoft/vscode/blob/6eaf748366a48a5b6f9645cfa5e2257f51a027a6/src/vs/workbench/contrib/preferences/browser/preferences.contribution.ts#L230-L240

Reply:
It seems that UI behavior is not consistent between `registerCommandAndKeybindingRule` and `Action + registerWorkbenchAction`.

In this screenshot, the first command is registered using `registerCommandAndKeybindingRule`, while the last to be `Action` and `registerWorkbenchAction`.

![image](https://user-images.githubusercontent.com/20227484/71586196-57e39000-2b54-11ea-94f8-296bb57cd401.png)

Through `registerWorkbenchAction`, the last command has an alias, and can be seen in command palette:

![image](https://user-images.githubusercontent.com/20227484/71586601-c543f080-2b55-11ea-9842-4a61e95235fc.png)

Command registered by `registerCommandAndKeybindingRule` does not have an alias and not get shown in command palette.

Reply:
For `openGlobalKeybindings`, it has alias and keybindings:
![image](https://user-images.githubusercontent.com/20227484/71586652-ec9abd80-2b55-11ea-84c4-4c8990736882.png)

and can be shown in command palette:
![image](https://user-images.githubusercontent.com/20227484/71586681-020fe780-2b56-11ea-9396-0774aadabd1d.png)

So, if we use `registerCommandAndKeybindingRule` to handle args, do we need to follow the implementation of settings editor to provide 2 `openGlobalKeybindings` and `openGlobalKeybindings2` commands?

Reply:
No. You can use `MenuRegistry` after using `egisterCommandAndKeybindingRule` to show the title. See 

https://github.com/microsoft/vscode/blob/806f1e51428aeaebe67974f1a34ba61c1cab445f/src/vs/workbench/contrib/preferences/browser/preferences.contribution.ts#L514-L522

Reply:
Thanks for pointing out it, I will have a try!

Reply:
I'm not sure what the status of this is, are you still working on it @Vigilans?